pkg/resourcemanager/controller/health/progressing/reconciler.go

```diff
@@ -131,7 +131,12 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, mr *resourc
 			return reconcile.Result{}, err
 		}
 
-		if progressing, description := checkProgressing(obj); progressing {
+		progressing, description, err := r.checkProgressing(ctx, obj)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if progressing {
```

Comment:
Not sure if we want it to be that short, but we could change it to:

```suggestion
		if progressing, description, err := r.checkProgressing(ctx, obj); err != nil {
			return reconcile.Result{}, err
		} else if progressing {
```